src/App.tsx

```diff
@@ -15,47 +17,124 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  NONE,
+  ALPHABET,
+  LENGTH,
+}
+
+type ReorderOptions = {
+  sortType: SortType,
+  isReversed: boolean,
+};
+
+export function getReorderedGoods(
+  goods: string[],
+  { sortType, isReversed }: ReorderOptions,
+) {
+  const visibleGoods = [...goods];
+
+  if (sortType) {
+    visibleGoods.sort((goodA, goodB) => {
+      switch (sortType) {
+        case SortType.ALPHABET:
+          return goodA.localeCompare(goodB);
+
+        case SortType.LENGTH:
+          return goodA.length - goodB.length;
+
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (isReversed) {
+    visibleGoods.reverse();
+  }
+
+  return visibleGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortType, setSortType] = useState(SortType.NONE);
+  const [isReversed, setIsReversed] = useState(false);
+  const [isVisible, setIsVisible] = useState(false);
```

Comment:
it is redundant state, you can calculate isVisible 
`const isVisible = useMemo(() => isReversed || sortType !== SortType.NONE, [isReversed, sortType])`